packages/react-ui/src/features/pieces/lib/pieces-hook.ts

```diff
@@ -29,6 +29,9 @@ export type StepMetadata = {
   displayName: string;
   logoUrl: string;
   description: string;
+  pieceName: string;
```

Comment:
Why these returns pieceName / pieceVersion to begin with?

I think if you are looking for pieces, then it's better to useMultiplePieces and these return the full model from the backend.